test/tests/functional/pbs_server_periodic_hook.py

```diff
@@ -70,17 +70,19 @@ def get_timestamp(self, msg):
         return int(a[1])
 
     def check_next_occurances(self, count, freq,
-                              hook_run_time, check_for_hook_end):
+                              hook_run_time, check_for_hook_end, alarm=0):
         """
         Helper function to check the occurances of hook by matching their
         logs in server logs.
         It needs 4 arguments:
-        - count			to know how many times to repeat
-                                checking these logs
-        - freq			is the frequency set in pbs server to run this hook
-        - hook_run_time		is the amount of time hook takes to run.
-        - check_for_hook_end	If it is true then the functions checks for
-                                hook end messages.
+        - count                     to know how many times to repeat
+                                    checking these logs
+        - freq                      is the frequency set in pbs server to run
+                                    this hook
+        - hook_run_time             is the amount of time hook takes to run.
+        - check_for_hook_end        If it is true then the functions checks for
+                                    hook end messages.
+        - alarm                     hook alarm
```

Comment:
If this is the hook alarm, don't you need to set it via a manager() call?

Reply:
The alarm is set by the caller while creating and importing the hook.